### PR TITLE
Replace plexus app events with goodies event bus

### PR DESCRIPTION
**Pull request by adreghiciu**
Created a new compatibility module (nexus-compat) that is including:
- the content of plexus app events api
- EventBus based implementation of ApplicationEventMulticaster
- the "old" / Event based events from sonatype security
- A shim between security new ovens and the old ones
- Replace all places where ApplicationEventMulticaster was used with EventBus
- Replace EventListener with event specific handlers

Depends on https://github.com/sonatype/security/pull/20

### CODE REVIEW

**Comment by adreghiciu** [comment]
CI voted +1 https://builds.sonatype.org/job/nexus-oss-its-feature/485/

**Comment by jdillon** [comment]
I look at this change a bit more today, though before we merge this I think we should get some load put under this in a test environment to see how it behaves.

**Comment by adreghiciu** [comment]
Another successful build with latest changes: https://builds.sonatype.org/view/nexus-features/job/nexus-oss-its-feature/487/

**Comment by cstamas** [comment]
+1 with same consideration as @jdillon has. @peterlynch any news?

**Comment by peterlynch** [comment]
@cstamas stressing this is being worked on today

File: nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/repository/AbstractRepository.java [hunk 1]
```diff
@@ -1257,6 +1258,10 @@ public void removeFromNotFoundCache( ResourceStoreRequest request )
     /**
      * Check conditions, such as availability, permissions, etc.
      *
+<<<<<<< HEAD
+     * @param request the request
+=======
+>>>>>>> refs/heads/master
```
Comment (nabcos) [inline nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/repository/AbstractRepository.java:1264]: unresolved merge conflict

**Pull request merged**